spec/unit/settings/server_list_setting_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'spec_helper'
+require 'puppet/settings/server_list_setting'
+
+describe Puppet::Settings::ServerListSetting do
+
+    it "prints strings as strings" do
+      settings = Puppet::Settings.new
+      settings.define_settings(:main, neptune: {type: :server_list, desc: 'list of servers'})
+      server_list_setting = settings.setting(:neptune)
+      expect(server_list_setting.print("jupiter,mars")).to eq("jupiter,mars")
+    end
+
+    it "prints arrays as strings" do
+      settings = Puppet::Settings.new
+      settings.define_settings(:main, neptune: {type: :server_list, desc: 'list of servers'})
+      server_list_setting = settings.setting(:neptune)
+      expect(server_list_setting.print([["main", 1234],["production", 8140]])).to eq("main:1234,production:8140")
+      expect(server_list_setting.print([])).to eq("")
```

Comment:
Will there ever be a case of a structure reaching print that is not a complete array of arrays, like: "myserver:123,someotherservername"?

Reply:
If `print` is going to get an array, from a string like that, the default port should have already been populated, so `["someotherservername", nil]` should never happen, if that's what you're asking about?